source/components/n2o_component.cpp

```diff
@@ -133,27 +133,29 @@ void N2OComponent::run( const double runToDate ) throw ( h_exception ) {
     
 	H_ASSERT( !core->inSpinup() && runToDate-oldDate == 1, "timestep must equal 1" );
    
-    // modified from Ward and Mahowald, 2014
+    // Approach modified from Ward and Mahowald, 2014, 10.5194/acp-14-12701-2014
     double previous_n2o = N0.value( U_PPBV_N2O );
 
-    if (runToDate!=oldDate)
-    {
+    if (runToDate != oldDate) {
       previous_n2o = N2O.get( oldDate );
     }
 
-    //Lifetime varys based on N2O concentrations
-    TAU_N2O.set( runToDate, unitval( TN2O0.value( U_YRS ) * ( pow( previous_n2o/N0.value( U_PPBV_N2O ),-0.05 ) ), U_YRS ) ); 
+    // Decay constant varies based on N2O concentrations
+    // This is Eq. B8 in Ward and Mahowald, 2014
+    TAU_N2O.set( runToDate, unitval( TN2O0.value( U_YRS ) * ( pow( previous_n2o /N0.value( U_PPBV_N2O ), -0.05 ) ), U_UNITLESS ) );
     
-    //varying natural emissions from preindustrial to 2000.  After 2000, we hold natural emissions constant
-       
-    const double current_n2oem = N2O_emissions.get( runToDate ).value( U_TG_N2O ) + N2ON_emissions.get( runToDate ).value( U_TG_N2O ); 
+    // Current emissions are the sum of natural and anthropogenic sources
+    const double current_n2oem = N2O_emissions.get( runToDate ).value( U_TG_N ) + N2O_natural_emissions.get( runToDate ).value( U_TG_N );
     
-    double dN2O = current_n2oem/UC_N2O - previous_n2o/TAU_N2O.get( runToDate ).value( U_YRS );
+    // This calculation follows Eq. B7 in Ward and Mahowald 2014
+    const double dN2O = current_n2oem / UC_N2O - previous_n2o / TAU_N2O.get( runToDate ).value( U_UNITLESS );
```

Comment:
Is `TAU_N2O` really unitless?  It seems like the kind of thing that should be in years.  Indeed, it was in years before.  Why the change?

Reply:
Mmm...looking back at the at the Ward et al. equation, I think you're right. Thanks.